packages/polling-controller/src/PollingController.ts

```diff
@@ -138,17 +153,20 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
      *
      * @param networkClientId - The networkClientId to listen for polling complete events
      * @param callback - The callback to execute when polling is complete
+     * @param options - The options used to group the polling events
      */
     onPollingCompleteByNetworkClientId(
       networkClientId: NetworkClientId,
       callback: (networkClientId: NetworkClientId) => void,
+      options: object = {},
     ) {
-      if (this.#callbacks.has(networkClientId)) {
-        this.#callbacks.get(networkClientId)?.add(callback);
+      const id = getKey(networkClientId, options);
+      if (this.#callbacks.has(id)) {
+        this.#callbacks.get(id)?.add(callback);
       } else {
         const set = new Set<typeof callback>();
         set.add(callback);
-        this.#callbacks.set(networkClientId, set);
+        this.#callbacks.set(id, set);
       }
```

Comment:
`?.` signals that we are doing nothing when the value we're looking for isn't there. However, that's unnecessary, as we just checked for the value. To remove the noise: Should we change this to:

``` suggestion
      const callbacks = this.#callbacks.get(id);

      if (callbacks === undefined) {
        const set = new Set<typeof callback>();
        set.add(callback);
        this.#callbacks.set(id, set);
      } else {
        callbacks.add(callback);
      }
```